lms/djangoapps/program_enrollments/tasks.py

```diff
@@ -0,0 +1,39 @@
+""" Tasks for program enrollments """
+from datetime import timedelta
+import logging
+from celery import task
+from celery_utils.logged_task import LoggedTask
+from django.utils import timezone
+
+from lms.djangoapps.program_enrollments.models import ProgramEnrollment
+
+log = logging.getLogger(__name__)
+
+
+@task(base=LoggedTask)
+def expire_waiting_enrollments(expiration_days):
+    """
+    Remove all program_enrollments and related program_course_enrollments for enrollments
```

Comment:
I'd tweak the wording on this to make it clearer that it's the program_enrollments whose age matters, and the related program_course_enrollments will be wiped regardless of their age.